### ISSUE TITLE
2.16.1 libtensorflow binary

### ISSUE THREAD

**Issue by yutkin** [tensorflow/tensorflow#63384]
### Issue type

Support

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

2.16.1

### Custom code

No

### OS platform and distribution

Linux

### GPU model and memory

Yes

### Current behavior?

Hi! 

Tensorflow 2.16.1 has been [released](https://github.com/tensorflow/tensorflow/releases/tag/v2.16.1) recently. However, the latest archive with the `libtensorflow` on the official website [is still 2.15](https://www.tensorflow.org/install/lang_c). Where can I get the latest 2.16.1 `libtensorflow` with GPU support for Linux?

### Standalone code to reproduce the issue

```shell
-
```

**Comment by mihaimaruseac** [comment]
CC: @MichaelHudgins @Venkat6871 @learning-to-play 

Is this because the libternsorflow build jobs have been discontinued?

**Comment by wamuir** [comment]
Additionally, the [libtensorflow nightly archive](https://storage.googleapis.com/libtensorflow-nightly?prefix=latest/libtensorflow-cpu-) also has not been updated for about sixty days.  And a bit longer for the [protos archive](https://storage.googleapis.com/libtensorflow-nightly?prefix=latest/libtensorflow_proto) and a few other artifacts

**Comment by MichaelHudgins** [comment]
These ones have not been discontinued at this time, it looks like an issue with some internal modifications not moving the built artifacts from a staging location to the final gcs bucket.

**Comment by MichaelHudgins** [comment]
Until documentation is corrected you can find it at 
https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow-gpu-linux-x86_64.tar.gz
https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow-cpu-linux-x86_64.tar.gz

**Comment by Venkat6871** [comment]
Hi **@yutkin** ,
TensorFlow 2.16.1 introduced a new installation method for Linux that allows installing the necessary NVIDIA CUDA libraries through pip. This means you can use pip install tensorflow[gpu] (assuming you have the NVIDIA driver already installed) to get TensorFlow with GPU support, even if there's no separate libtensorflow binary.

Thank you!

**Comment by MichaelHudgins** [comment]
Hi @Venkat6871 , this is not an issue with a pip installation but rather one from the libtensorflow libraries not being in the expected locaiton.

**Comment by karllessard** [comment]
@MichaelHudgins , I can't find the Windows libtensorflow under the new location, I've tried this, anything it wrong with that URL?

[https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow-cpu-windows-x86_64-2.16.1.zip](https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow-cpu-windows-x86_64-2.16.1.zip)

Can't find it in the previous location neither, last one is 2.15.0

**Comment by karllessard** [comment]
List the content of the bucket, looks like Windows is nowhere to be seen. Has the libtensorflow build for Windows been deprecated? (hopefully not...)
```
gcloud storage ls gs://tensorflow/versions/2.16.1
gs://tensorflow/versions/2.16.1/libtensorflow-cpu-darwin-arm64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow-cpu-darwin-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow-cpu-linux-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow-gpu-linux-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow_jni-cpu-darwin-arm64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow_jni-cpu-darwin-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow_jni-cpu-linux-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/libtensorflow_jni-gpu-linux-x86_64.tar.gz
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp310-cp310-macosx_10_15_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp310-cp310-macosx_12_0_arm64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp311-cp311-macosx_10_15_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp311-cp311-macosx_12_0_arm64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp312-cp312-macosx_10_15_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp312-cp312-macosx_12_0_arm64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp39-cp39-macosx_10_15_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp39-cp39-macosx_12_0_arm64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
gs://tensorflow/versions/2.16.1/tensorflow-2.16.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_cpu-2.16.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_cpu-2.16.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_cpu-2.16.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_cpu-2.16.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_tpu-2.16.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_tpu-2.16.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_tpu-2.16.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
gs://tensorflow/versions/2.16.1/tensorflow_tpu-2.16.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```

**Comment by adamcrume** [comment]
This is blocking the TensorFlow Rust release.

**Comment by karllessard** [comment]
> This is blocking the TensorFlow Rust release.

Same for TensorFlow Java

**Comment by nitins17** [comment]
Thanks for creating this issue. It looks like the Windows CI build uploaded to the incorrect location. I have moved it to be under `gs://tensorflow/versions/2.16.1`. 

https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow-cpu-windows-x86_64.zip
https://storage.googleapis.com/tensorflow/versions/2.16.1/libtensorflow_jni-cpu-windows-x86_64.zip

**Comment by marcellobarile** [comment]
@nitins17 what about the GPU version? Thanks

**Comment by nitins17** [comment]
> When is this expected to be resolved?

My conservative guess would be by end of next week. The change to update the links on the website is in progress internally, just waiting on the final approval.

**Comment by yutkin** [comment]
> > When is this expected to be resolved?
> 
> My conservative guess would be by end of next week. The change to update the links on the website is in progress internally, just waiting on the final approval.

Hi! Are there any updates?

**Comment by nitins17** [comment]
Hmm, the [commit has been merged](https://github.com/tensorflow/docs/commit/29fd0f4fe59582b65ec33102385189ea3a9ba9c5) but I'm not sure why it does not show up on the website. @MarkDaoust would you know?

@yutkin In the meantime, you can use the preview in https://github.com/tensorflow/docs/blob/master/site/en/install/lang_c.ipynb to get the links to the latest archives.

**Comment by yutkin** [comment]
The docs still point to the 2.15 version.

**Comment by yutkin** [comment]
2.16.2 is here. Do you have any update on the issue?

**Comment by rtg0795** [comment]
I am following up with docs team to get this fixed.

**Comment by yutkin** [comment]
Hi, team! Any update? It is still 2.15, the latest binaries in the docs. Although 2.18 was released recently.

**Comment by nitins17** [comment]
Sorry about the long delay. The latest libtensorflow binaries can now be found at https://www.tensorflow.org/install/lang_c.

**Comment by wo80** [comment]
This is not an issue perse, so I'm using this thread to ask:

I am currently using libtensorflow-cpu-windows version 2.15. The dll has a size of 229MB, the lib file has 52MB. Now the latest published version 2.18 has a dll size of 909MB and the lib file 9MB. Can anyone explain this change in sizes?

**Comment by nitins17** [comment]
Thanks for letting us know. That is certainly a very big jump and is something we need to look into. As for how we build the archive, see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/ci_build/rel/windows/cpu_libtensorflow.bat
